glass/core/algorithm.py

```diff
@@ -51,8 +56,11 @@ def nnls(
           Chemometrics, 11, 393-401.
 
     """
-    a = np.asanyarray(a)
-    b = np.asanyarray(b)
+    if a.__array_namespace__() != b.__array_namespace__():
+        msg = "input arrays should belong to the same array library"
+        raise ValueError(msg)
+
+    xp = a.__array_namespace__()
```

Comment:
Was talking to @matt-graham and he wondered if we should be using [array-api-compat](https://github.com/data-apis/array-api-compat) here. That should solve the `cupy` issue. I believe only `NumPy` and `JAX` are stable at this point.

Reply:
The CuPy support will become official in their next major release (`v14`) where they will start supporting NumPy `v2`. I hope they push the release in the next 3 months, so that we don't have to experiment with [array-api-compat](https://github.com/data-apis/array-api-compat).